test/node/src/with-schema.test.ts

```diff
@@ -517,6 +520,72 @@ for (const dialect of DIALECTS.filter(
       })
     })
 
+    describe('raw sql', () => {
+      let dbWithSchema: Kysely<any>
+      const subsuites = [
+        ['sql', () => sql],
+        [
+          'eb',
+          () => createExpressionBuilder<any, any>(dbWithSchema.getExecutor()),
+        ],
+      ] as unknown as [string, () => Sql][]
+
+      before(() => {
+        dbWithSchema = ctx.db.withSchema('mammals')
+      })
+
+      it('should not add schema for raw sql', async () => {
+        const query =
+          sql`"pet" "mammals"."pet" "id" "pet"."id" "mammals"."pet"."id" pet mammals.pet id pet.id mammals.pet.id`.compile(
+            dbWithSchema
+          )
+
+        expect(query.sql).to.equal(
+          '"pet" "mammals"."pet" "id" "pet"."id" "mammals"."pet"."id" pet mammals.pet id pet.id mammals.pet.id'
+        )
+      })
+
+      subsuites.forEach(([name, builder]) => {
+        describe(`${name}.table`, () => {
+          it('should add schema', () => {
```

Comment:
Add a test case like this:

```ts
sql`
  select * from person as p
  inner join pets on pets.owner_id = ${ref.ref('p.id')}
`
```

I think the current code would incorrectly add schema to `mammals.p.id` 🤔

It might be impossible to handle this correctly since we don't know how `ref` and `table` are used in raw SQL.

Reply:
Damn, got some rethinking to do. Great catch! 💪